pkg/rancher-desktop/window/index.ts

```diff
@@ -218,6 +218,57 @@ function createView() {
 
   if (currentExtension?.id) {
     webPreferences.partition = `persist:rdx-${ currentExtension.id }`;
+    const webRequest = Electron.session.fromPartition(webPreferences.partition).webRequest;
+
+    webRequest.onBeforeSendHeaders((details, callback) => {
+      const source = details.webContents?.getURL() ?? '';
+      const requestHeaders = { ...details.requestHeaders };
+
+      if (isInternalURL(source)) {
+        // If the request is coming from the extension, remove the Origin: header
+        // because it has x-rd-extension:// nonsense (relative to the server).
+        delete requestHeaders.Origin;
+      }
+      callback({ requestHeaders });
+    });
+
+    webRequest.onHeadersReceived((details, callback) => {
+      const sourceURL = details.webContents?.getURL() ?? '';
+
+      if (!isInternalURL(sourceURL)) {
+        // Do not rewrite requests from outside the extension (e.g. iframe).
+        callback({});
+
+        return;
+      }
+
+      // Insert (or overwrite) CORS headers to pretend this was allowed.  While
+      // ideally we can just disable `webSecurity` instead, that seems to break
+      // the preload script (which breaks the extension APIs).
+      const responseHeaders: Record<string, string|string[]> = { ...details.responseHeaders };
+      const { compare } = new Intl.Collator('en', { sensitivity: 'accent' });
```

Comment:
I assume this means `é` and `e` are different, but `A` and `a` are treated the same? nit; it would be nice to document.

Reply:
Yes; but HTTP headers only support standard ASCII anyway.